pkg/api/v1/types.go

```diff
@@ -541,3 +545,74 @@ func (vl *VirtualMachineReplicaSetList) GetObjectKind() schema.ObjectKind {
 func (vl *VirtualMachineReplicaSetList) GetListMeta() meta.List {
 	return &vl.ListMeta
 }
+
+// OfflineVirtualMachine handles the VirtualMachines that are not running
+// or are in a stopped state
+// The OfflineVirtualMachine contains the template to create the
+// VirtualMachine. It also mirrors the running state of the created
+// VirtualMachine in its status.
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type OfflineVirtualMachine struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   OfflineVirtualMachineSpec   `json:"spec,omitempty"`
+	Status OfflineVirtualMachineStatus `json:"status,omitempty"`
```

Comment:
All fields should have a comment which describes them. Please go over all nested structs and make sure that they have a description.